Source/Blazorise/Utilities/IO/FileEntry.cs

```diff
@@ -57,6 +58,21 @@ public Task Cancel()
 
         return Task.CompletedTask;
     }
+    
+    ~FileEntry()
+    {
+        try
+        {
+            Owner.RemoveFileEntry( Id );//when the owner is alive this will work
```

Comment:
Still, the problem might be that we're calling asynchronous code from the finalizer, which is synchronous. This might lead to deadlock at some point. Can we not use finalizer at all for this? Can we JS dispose of fileentry when it is removed from a list somewhere else?

Reply:
I don't like using a finalizer either. If there were another non-API-breaking solution, I would use it, but I don't see one. Do you have any ideas?  

We can't reliably determine when the C# `FileEntry` object is no longer needed, so the finalizer is the only place to dispose of the JS resources as well.  

I don't think the current solution would cause deadlocks because the async code in the finalizer runs in a fire-and-forget fashion. It doesn't wait for anything, so there’s no point where it could get locked.

I would still prefer to break the api with  `WriteToStreamIfAvailableAsync` or `TryWriteToStreamAsync` and removing the blob from js after the `OnImageUploadChanged` has been invoked.